### DISCUSSION TITLE
URL rejected: say by who. Closing connection: only say if there was one to close

### DISCUSSION THREAD

**Discussion by jidanni** [curl/curl#11788] [Ideas]
With -v output like
```
* processing: http://localhost/?data=(relation["wikipedia"="en:Winnetka, Illinois"];(._;>;);out body;
* URL rejected: Malformed input to a URL function
* Closing connection
```
1. It is not clear who rejected the URL. Curl, or the server.
2. It must be the latter, we think, as we see "Closing connection", so there must have been a connection to the remote.

First, please say
* URL rejected _by curl_: Malformed input to a URL function

Second, please only say "Closing connection" if there is a connection to close.
Proof:
```
$ curl -v localhost
* processing: localhost
*   Trying [::1]:80...
* connect to ::1 port 80 failed: Connection refused
*   Trying 127.0.0.1:80...
* connect to 127.0.0.1 port 80 failed: Connection refused
* Failed to connect to localhost port 80 after 1 ms: Couldn't connect to server
* Closing connection
curl: (7) Failed to connect to localhost port 80 after 1 ms: Couldn't connect to server
```

**Comment by jay** [comment]
you may have to infer from more than a single verbose message to understand the context. in my opinion those messages are acceptable as is.

**Reply by jidanni** [reply]
That's what I'm saying: the messages could be clearer on their own, so we wouldn't need to dig deeper.

We could see it right up off the bat that the same problem would occur without even needing a destination...

Which brings up another question: what destination should I use if I just want to test the validity of my URL, and I'm offline, and I don't even have a server on localhost?

Or is the best I can do is just use the word localhost anyway?

**Reply by bagder** [reply]
>  I just want to test the validity of my URL

[trurl](https://curl.se/trurl)
